src/MagnumPlugins/BasisImageConverter/BasisImageConverter.cpp

```diff
@@ -46,30 +46,90 @@
+
+    if(yFlip)
+        dst = dst.flipped<0>();
+
+    /* basis image is always RGBA, fill in alpha if necessary */
+    if(channelCount == 4) {
+        const auto src = image3D.pixels<Math::Vector<4, T>>()[slice];
+        for(std::size_t y = 0; y != src.size()[0]; ++y)
+            for(std::size_t x = 0; x != src.size()[1]; ++x)
+                dst[y][x] = src[y][x];
+
+    } else if(channelCount == 3) {
+        const auto src = image3D.pixels<Math::Vector<3, T>>()[slice];
+        for(std::size_t y = 0; y != src.size()[0]; ++y)
+            for(std::size_t x = 0; x != src.size()[1]; ++x)
+                dst[y][x] = Math::Vector3<T>{src[y][x]}; /* Alpha implicitly opaque */
+
+    } else if(channelCount == 2) {
+        const auto src = image3D.pixels<Math::Vector<2, T>>()[slice];
+        /* If the user didn't specify a custom swizzle, assume they want
+           the two channels compressed in separate slices, R in RGB and G
+           in Alpha. This significantly improves quality. */
+        if(!customSwizzle)
+            for(std::size_t y = 0; y != src.size()[0]; ++y)
+                for(std::size_t x = 0; x != src.size()[1]; ++x)
+                    dst[y][x] = Math::gather<'r', 'r', 'r', 'g'>(src[y][x]);
+        else
+            for(std::size_t y = 0; y != src.size()[0]; ++y)
+                for(std::size_t x = 0; x != src.size()[1]; ++x)
+                    dst[y][x] = Math::Vector3<T>::pad(src[y][x]); /* Alpha implicitly opaque */
+
+    } else if(channelCount == 1) {
+        const auto src = image3D.pixels<Math::Vector<1, T>>()[slice];
+        /* If the user didn't specify a custom swizzle, assume they want
+           a gray-scale image. Alpha is always implicitly opaque. */
+        if(!customSwizzle)
+            for(std::size_t y = 0; y != src.size()[0]; ++y)
+                for(std::size_t x = 0; x != src.size()[1]; ++x)
+                    dst[y][x] = Math::gather<'r', 'r', 'r'>(src[y][x]);
+        else
+            for(std::size_t y = 0; y != src.size()[0]; ++y)
+                for(std::size_t x = 0; x != src.size()[1]; ++x)
+                    dst[y][x] = Math::Vector3<T>::pad(src[y][x]); /* Alpha implicitly opaque */
+
+    } else CORRADE_INTERNAL_ASSERT_UNREACHABLE(); /* LCOV_EXCL_LINE */
+}
+
 template<UnsignedInt dimensions> Containers::Optional<Containers::Array<char>> convertLevelsToData(Containers::ArrayView<const BasicImageView<dimensions>> imageLevels, const Utility::ConfigurationGroup& configuration, ImageConverterFlags flags, BasisImageConverter::Format fileFormat) {
     /* Check input */
     const PixelFormat pixelFormat = imageLevels.front().format();
-    bool isSrgb;
+    bool isSrgb = false;
+    bool isHdr = false;
     switch(pixelFormat) {
         case PixelFormat::RGBA8Unorm:
         case PixelFormat::RGB8Unorm:
         case PixelFormat::RG8Unorm:
         case PixelFormat::R8Unorm:
-            isSrgb = false;
```

Comment:
I'd rather keep this here, to not have some of those accidentally left at default in some codepaths. Latest GCC and/or Clang warns (or they do in Release builds at least) if the variable isn't initialized in all codepaths, which is a good hint that something went wrong. With the default initialization an accident would be silent.